compilation fails, many symbols not found in scope

Compilation log created by media-autobuild suite:

```
CFLAGS: -mthreads -mtune=generic -O2 -pipe
LDFLAGS: -pipe -static-libgcc -static-libstdc++
/opt/cargo/bin/cargo.exe build --release --target=i686-pc-windows-gnu --jobs=4
   Compiling rav1e v0.1.0 (E:\MABS\build\rav1e-git)
error[E0425]: cannot find value `CDEF_FILTER_FNS` in this scope
  --> src\asm\x86\cdef.rs:76:13
   |
76 |       match CDEF_FILTER_FNS[cpu.as_index()][decimate_index(xdec, ydec)] {
   |             ^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `CDEF_FILTER_HBD_FNS` in this scope
  --> src\asm\x86\cdef.rs:93:13
   |
93 |       match CDEF_FILTER_HBD_FNS[cpu.as_index()][decimate_index(xdec, ydec)] {
   |             ^^^^^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `SAD_FNS` in this scope
   --> src\asm\x86\dist.rs:123:28
    |
123 |     PixelType::U8 => match SAD_FNS[cpu.as_index()][to_index(bsize)] {
    |                            ^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
    |
10  | use crate::asm::x86::dist::x86_64::SAD_FNS;
    |

error[E0425]: cannot find value `SAD_HBD_FNS` in this scope
   --> src\asm\x86\dist.rs:134:29
    |
134 |     PixelType::U16 => match SAD_HBD_FNS[cpu.as_index()][to_index(bsize)] {
    |                             ^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
    |
10  | use crate::asm::x86::dist::x86_64::SAD_HBD_FNS;
    |

error[E0425]: cannot find value `SATD_FNS` in this scope
   --> src\asm\x86\dist.rs:166:28
    |
166 |     PixelType::U8 => match SATD_FNS[cpu.as_index()][to_index(bsize)] {
    |                            ^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
    |
10  | use crate::asm::x86::dist::x86_64::SATD_FNS;
    |

error[E0425]: cannot find value `SATD_HBD_FNS` in this scope
   --> src\asm\x86\dist.rs:177:29
    |
177 |     PixelType::U16 => match SATD_HBD_FNS[cpu.as_index()][to_index(bsize)] {
    |                             ^^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
    |
10  | use crate::asm::x86::dist::x86_64::SATD_HBD_FNS;
    |

error[E0425]: cannot find function `_mm_cvtsi64_si128` in this scope
  --> src\asm\x86\ec.rs:29:39
   |
29 |     _mm_sub_epi16(v, _mm_srl_epi16(v, _mm_cvtsi64_si128(rate as i64)));
   |                                       ^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `_mm_cvtsi32_si128`

error[E0425]: cannot find function `_mm_cvtsi64_si128` in this scope
  --> src\asm\x86\ec.rs:35:7
   |
35 |       _mm_cvtsi64_si128(rate as i64),
   |       ^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `_mm_cvtsi32_si128`

error[E0425]: cannot find value `PUT_FNS` in this scope
   --> src\asm\x86\mc.rs:106:13
    |
106 |       match PUT_FNS[cpu.as_index()][get_2d_mode_idx(mode_x, mode_y)] {
    |             ^^^^^^^ not found in this scope

error[E0425]: cannot find value `PUT_HBD_FNS` in this scope
   --> src\asm\x86\mc.rs:123:13
    |
123 |       match PUT_HBD_FNS[cpu.as_index()][get_2d_mode_idx(mode_x, mode_y)] {
    |             ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `PREP_FNS` in this scope
   --> src\asm\x86\mc.rs:173:13
    |
173 |       match PREP_FNS[cpu.as_index()][get_2d_mode_idx(mode_x, mode_y)] {
    |             ^^^^^^^^ not found in this scope

error[E0425]: cannot find value `PREP_HBD_FNS` in this scope
   --> src\asm\x86\mc.rs:189:13
    |
189 |       match PREP_HBD_FNS[cpu.as_index()][get_2d_mode_idx(mode_x, mode_y)] {
    |             ^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `AVG_FNS` in this scope
   --> src\asm\x86\mc.rs:226:28
    |
226 |     PixelType::U8 => match AVG_FNS[cpu.as_index()] {
    |                            ^^^^^^^ not found in this scope

error[E0425]: cannot find value `AVG_HBD_FNS` in this scope
   --> src\asm\x86\mc.rs:239:29
    |
239 |     PixelType::U16 => match AVG_HBD_FNS[cpu.as_index()] {
    |                             ^^^^^^^^^^^ not found in this scope

warning: ignoring -C extra-filename flag due to -o flag

error: aborting due to 14 previous errors

For more information about this error, try `rustc --explain E0425`.
error: Could not compile `rav1e`.

To learn more, run the command again with --verbose.
```

--- a/src/asm/x86/cdef.rs
+++ b/src/asm/x86/cdef.rs
@@ -14,9 +14,6 @@ use crate::frame::*;
 use crate::tiling::PlaneRegionMut;
 use crate::util::*;
 
-#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
-use x86_64::*;
-
 type CdefFilterFn = unsafe extern fn(
   dst: *mut u8,
   dst_stride: isize,
@@ -120,43 +117,38 @@ pub unsafe fn cdef_filter_block<T: Pixel>(
   }
 }
 
-#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
-mod x86_64 {
-  use super::*;
+extern {
+  fn rav1e_cdef_filter_4x4_avx2(
+    dst: *mut u8, dst_stride: isize, tmp: *const u16, tmp_stride: isize,
+    pri_strength: i32, sec_strength: i32, dir: i32, damping: i32,
+  );
 
-  extern {
-    fn rav1e_cdef_filter_4x4_avx2(
-      dst: *mut u8, dst_stride: isize, tmp: *const u16, tmp_stride: isize,
-      pri_strength: i32, sec_strength: i32, dir: i32, damping: i32,
-    );
+  fn rav1e_cdef_filter_4x8_avx2(
+    dst: *mut u8, dst_stride: isize, tmp: *const u16, tmp_stride: isize,
+    pri_strength: i32, sec_strength: i32, dir: i32, damping: i32,
+  );
 
-    fn rav1e_cdef_filter_4x8_avx2(
-      dst: *mut u8, dst_stride: isize, tmp: *const u16, tmp_stride: isize,
-      pri_strength: i32, sec_strength: i32, dir: i32, damping: i32,
-    );
-
-    fn rav1e_cdef_filter_8x8_avx2(
-      dst: *mut u8, dst_stride: isize, tmp: *const u16, tmp_stride: isize,
-      pri_strength: i32, sec_strength: i32, dir: i32, damping: i32,
-    );
-  }
-
-  static CDEF_FILTER_FNS_AVX2: [Option<CdefFilterFn>; 4] = {
-    let mut out: [Option<CdefFilterFn>; 4] = [None; 4];
-    out[decimate_index(1, 1)] = Some(rav1e_cdef_filter_4x4_avx2);
-    out[decimate_index(1, 0)] = Some(rav1e_cdef_filter_4x8_avx2);
-    out[decimate_index(0, 0)] = Some(rav1e_cdef_filter_8x8_avx2);
-    out
-  };
-
-  pub(crate) static CDEF_FILTER_FNS: [[Option<CdefFilterFn>; 4];
-    CpuFeatureLevel::len()] = {
-    let mut out: [[Option<CdefFilterFn>; 4]; CpuFeatureLevel::len()] =
-      [[None; 4]; CpuFeatureLevel::len()];
-    out[CpuFeatureLevel::AVX2 as usize] = CDEF_FILTER_FNS_AVX2;
-    out
-  };
-
-  pub(crate) static CDEF_FILTER_HBD_FNS: [[Option<CdefFilterHBDFn>; 4];
-    CpuFeatureLevel::len()] = [[None; 4]; CpuFeatureLevel::len()];
+  fn rav1e_cdef_filter_8x8_avx2(
+    dst: *mut u8, dst_stride: isize, tmp: *const u16, tmp_stride: isize,
+    pri_strength: i32, sec_strength: i32, dir: i32, damping: i32,
+  );
 }
+
+static CDEF_FILTER_FNS_AVX2: [Option<CdefFilterFn>; 4] = {
+  let mut out: [Option<CdefFilterFn>; 4] = [None; 4];
+  out[decimate_index(1, 1)] = Some(rav1e_cdef_filter_4x4_avx2);
+  out[decimate_index(1, 0)] = Some(rav1e_cdef_filter_4x8_avx2);
+  out[decimate_index(0, 0)] = Some(rav1e_cdef_filter_8x8_avx2);
+  out
+};
+
+pub(crate) static CDEF_FILTER_FNS: [[Option<CdefFilterFn>; 4];
+  CpuFeatureLevel::len()] = {
+  let mut out: [[Option<CdefFilterFn>; 4]; CpuFeatureLevel::len()] =
+    [[None; 4]; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::AVX2 as usize] = CDEF_FILTER_FNS_AVX2;
+  out
+};
+
+pub(crate) static CDEF_FILTER_HBD_FNS: [[Option<CdefFilterHBDFn>; 4];
+  CpuFeatureLevel::len()] = [[None; 4]; CpuFeatureLevel::len()];

--- a/src/asm/x86/dist.rs
+++ b/src/asm/x86/dist.rs
@@ -13,9 +13,6 @@ use crate::partition::BlockSize;
 use crate::tiling::*;
 use crate::util::*;
 
-#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
-use x86_64::*;
-
 type SadFn = unsafe extern fn(
   src: *const u8,
   src_stride: isize,
@@ -193,142 +190,137 @@ pub fn get_satd<T: Pixel>(
   dist
 }
 
-mod x86_64 {
-  use super::*;
+static SAD_FNS_SSE2: [Option<SadFn>; DIST_FNS_LENGTH] = {
+  let mut out: [Option<SadFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
 
-  static SAD_FNS_SSE2: [Option<SadFn>; DIST_FNS_LENGTH] = {
-    let mut out: [Option<SadFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
+  use BlockSize::*;
 
-    use BlockSize::*;
+  out[BLOCK_4X4 as usize] = Some(rav1e_sad4x4_sse2);
+  out[BLOCK_4X8 as usize] = Some(rav1e_sad4x8_sse2);
+  out[BLOCK_4X16 as usize] = Some(rav1e_sad4x16_sse2);
 
-    out[BLOCK_4X4 as usize] = Some(rav1e_sad4x4_sse2);
-    out[BLOCK_4X8 as usize] = Some(rav1e_sad4x8_sse2);
-    out[BLOCK_4X16 as usize] = Some(rav1e_sad4x16_sse2);
+  out[BLOCK_8X4 as usize] = Some(rav1e_sad8x4_sse2);
+  out[BLOCK_8X8 as usize] = Some(rav1e_sad8x8_sse2);
+  out[BLOCK_8X16 as usize] = Some(rav1e_sad8x16_sse2);
+  out[BLOCK_8X32 as usize] = Some(rav1e_sad8x32_sse2);
 
-    out[BLOCK_8X4 as usize] = Some(rav1e_sad8x4_sse2);
-    out[BLOCK_8X8 as usize] = Some(rav1e_sad8x8_sse2);
-    out[BLOCK_8X16 as usize] = Some(rav1e_sad8x16_sse2);
-    out[BLOCK_8X32 as usize] = Some(rav1e_sad8x32_sse2);
+  out[BLOCK_16X16 as usize] = Some(rav1e_sad16x16_sse2);
+  out[BLOCK_32X32 as usize] = Some(rav1e_sad32x32_sse2);
+  out[BLOCK_64X64 as usize] = Some(rav1e_sad64x64_sse2);
+  out[BLOCK_128X128 as usize] = Some(rav1e_sad128x128_sse2);
 
-    out[BLOCK_16X16 as usize] = Some(rav1e_sad16x16_sse2);
-    out[BLOCK_32X32 as usize] = Some(rav1e_sad32x32_sse2);
-    out[BLOCK_64X64 as usize] = Some(rav1e_sad64x64_sse2);
-    out[BLOCK_128X128 as usize] = Some(rav1e_sad128x128_sse2);
+  out
+};
 
-    out
-  };
+static SAD_FNS_AVX2: [Option<SadFn>; DIST_FNS_LENGTH] = {
+  let mut out: [Option<SadFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
 
-  static SAD_FNS_AVX2: [Option<SadFn>; DIST_FNS_LENGTH] = {
-    let mut out: [Option<SadFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
+  use BlockSize::*;
 
-    use BlockSize::*;
+  out[BLOCK_4X4 as usize] = Some(rav1e_sad4x4_sse2);
+  out[BLOCK_4X8 as usize] = Some(rav1e_sad4x8_sse2);
+  out[BLOCK_4X16 as usize] = Some(rav1e_sad4x16_sse2);
 
-    out[BLOCK_4X4 as usize] = Some(rav1e_sad4x4_sse2);
-    out[BLOCK_4X8 as usize] = Some(rav1e_sad4x8_sse2);
-    out[BLOCK_4X16 as usize] = Some(rav1e_sad4x16_sse2);
+  out[BLOCK_8X4 as usize] = Some(rav1e_sad8x4_sse2);
+  out[BLOCK_8X8 as usize] = Some(rav1e_sad8x8_sse2);
+  out[BLOCK_8X16 as usize] = Some(rav1e_sad8x16_sse2);
+  out[BLOCK_8X32 as usize] = Some(rav1e_sad8x32_sse2);
 
-    out[BLOCK_8X4 as usize] = Some(rav1e_sad8x4_sse2);
-    out[BLOCK_8X8 as usize] = Some(rav1e_sad8x8_sse2);
-    out[BLOCK_8X16 as usize] = Some(rav1e_sad8x16_sse2);
-    out[BLOCK_8X32 as usize] = Some(rav1e_sad8x32_sse2);
+  out[BLOCK_16X4 as usize] = Some(rav1e_sad16x4_avx2);
+  out[BLOCK_16X8 as usize] = Some(rav1e_sad16x8_avx2);
+  out[BLOCK_16X16 as usize] = Some(rav1e_sad16x16_avx2);
+  out[BLOCK_16X32 as usize] = Some(rav1e_sad16x32_avx2);
+  out[BLOCK_16X64 as usize] = Some(rav1e_sad16x64_avx2);
 
-    out[BLOCK_16X4 as usize] = Some(rav1e_sad16x4_avx2);
-    out[BLOCK_16X8 as usize] = Some(rav1e_sad16x8_avx2);
-    out[BLOCK_16X16 as usize] = Some(rav1e_sad16x16_avx2);
-    out[BLOCK_16X32 as usize] = Some(rav1e_sad16x32_avx2);
-    out[BLOCK_16X64 as usize] = Some(rav1e_sad16x64_avx2);
+  out[BLOCK_32X8 as usize] = Some(rav1e_sad32x8_avx2);
+  out[BLOCK_32X16 as usize] = Some(rav1e_sad32x16_avx2);
+  out[BLOCK_32X32 as usize] = Some(rav1e_sad32x32_avx2);
+  out[BLOCK_32X64 as usize] = Some(rav1e_sad32x64_avx2);
 
-    out[BLOCK_32X8 as usize] = Some(rav1e_sad32x8_avx2);
-    out[BLOCK_32X16 as usize] = Some(rav1e_sad32x16_avx2);
-    out[BLOCK_32X32 as usize] = Some(rav1e_sad32x32_avx2);
-    out[BLOCK_32X64 as usize] = Some(rav1e_sad32x64_avx2);
+  out[BLOCK_64X16 as usize] = Some(rav1e_sad64x16_avx2);
+  out[BLOCK_64X32 as usize] = Some(rav1e_sad64x32_avx2);
+  out[BLOCK_64X64 as usize] = Some(rav1e_sad64x64_avx2);
+  out[BLOCK_64X128 as usize] = Some(rav1e_sad64x128_avx2);
 
-    out[BLOCK_64X16 as usize] = Some(rav1e_sad64x16_avx2);
-    out[BLOCK_64X32 as usize] = Some(rav1e_sad64x32_avx2);
-    out[BLOCK_64X64 as usize] = Some(rav1e_sad64x64_avx2);
-    out[BLOCK_64X128 as usize] = Some(rav1e_sad64x128_avx2);
+  out[BLOCK_128X64 as usize] = Some(rav1e_sad128x64_avx2);
+  out[BLOCK_128X128 as usize] = Some(rav1e_sad128x128_avx2);
 
-    out[BLOCK_128X64 as usize] = Some(rav1e_sad128x64_avx2);
-    out[BLOCK_128X128 as usize] = Some(rav1e_sad128x128_avx2);
+  out
+};
 
-    out
-  };
+pub static SAD_FNS: [[Option<SadFn>; DIST_FNS_LENGTH];
+  CpuFeatureLevel::len()] = {
+  let mut out = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
 
-  pub static SAD_FNS: [[Option<SadFn>; DIST_FNS_LENGTH];
-    CpuFeatureLevel::len()] = {
-    let mut out = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::SSE2 as usize] = SAD_FNS_SSE2;
+  out[CpuFeatureLevel::SSSE3 as usize] = SAD_FNS_SSE2;
+  out[CpuFeatureLevel::AVX2 as usize] = SAD_FNS_AVX2;
 
-    out[CpuFeatureLevel::SSE2 as usize] = SAD_FNS_SSE2;
-    out[CpuFeatureLevel::SSSE3 as usize] = SAD_FNS_SSE2;
-    out[CpuFeatureLevel::AVX2 as usize] = SAD_FNS_AVX2;
+  out
+};
 
-    out
-  };
+static SAD_HBD_FNS_SSSE3: [Option<SadHBDFn>; DIST_FNS_LENGTH] = {
+  let mut out: [Option<SadHBDFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
 
-  static SAD_HBD_FNS_SSSE3: [Option<SadHBDFn>; DIST_FNS_LENGTH] = {
-    let mut out: [Option<SadHBDFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
+  use BlockSize::*;
 
-    use BlockSize::*;
+  out[BLOCK_4X4 as usize] = Some(rav1e_sad_4x4_hbd_ssse3);
+  out[BLOCK_16X16 as usize] = Some(rav1e_sad_16x16_hbd_ssse3);
 
-    out[BLOCK_4X4 as usize] = Some(rav1e_sad_4x4_hbd_ssse3);
-    out[BLOCK_16X16 as usize] = Some(rav1e_sad_16x16_hbd_ssse3);
+  out
+};
 
-    out
-  };
+pub(crate) static SAD_HBD_FNS: [[Option<SadHBDFn>; DIST_FNS_LENGTH];
+  CpuFeatureLevel::len()] = {
+  let mut out = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
 
-  pub(crate) static SAD_HBD_FNS: [[Option<SadHBDFn>; DIST_FNS_LENGTH];
-    CpuFeatureLevel::len()] = {
-    let mut out = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::SSSE3 as usize] = SAD_HBD_FNS_SSSE3;
+  out[CpuFeatureLevel::AVX2 as usize] = SAD_HBD_FNS_SSSE3;
 
-    out[CpuFeatureLevel::SSSE3 as usize] = SAD_HBD_FNS_SSSE3;
-    out[CpuFeatureLevel::AVX2 as usize] = SAD_HBD_FNS_SSSE3;
+  out
+};
 
-    out
-  };
+static SATD_FNS_AVX2: [Option<SatdFn>; DIST_FNS_LENGTH] = {
+  let mut out: [Option<SatdFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
 
-  static SATD_FNS_AVX2: [Option<SatdFn>; DIST_FNS_LENGTH] = {
-    let mut out: [Option<SatdFn>; DIST_FNS_LENGTH] = [None; DIST_FNS_LENGTH];
+  use BlockSize::*;
 
-    use BlockSize::*;
+  out[BLOCK_4X4 as usize] = Some(rav1e_satd_4x4_avx2);
+  out[BLOCK_8X8 as usize] = Some(rav1e_satd_8x8_avx2);
+  out[BLOCK_16X16 as usize] = Some(rav1e_satd_16x16_avx2);
+  out[BLOCK_32X32 as usize] = Some(rav1e_satd_32x32_avx2);
+  out[BLOCK_64X64 as usize] = Some(rav1e_satd_64x64_avx2);
+  out[BLOCK_128X128 as usize] = Some(rav1e_satd_128x128_avx2);
 
-    out[BLOCK_4X4 as usize] = Some(rav1e_satd_4x4_avx2);
-    out[BLOCK_8X8 as usize] = Some(rav1e_satd_8x8_avx2);
-    out[BLOCK_16X16 as usize] = Some(rav1e_satd_16x16_avx2);
-    out[BLOCK_32X32 as usize] = Some(rav1e_satd_32x32_avx2);
-    out[BLOCK_64X64 as usize] = Some(rav1e_satd_64x64_avx2);
-    out[BLOCK_128X128 as usize] = Some(rav1e_satd_128x128_avx2);
+  out[BLOCK_4X8 as usize] = Some(rav1e_satd_4x8_avx2);
+  out[BLOCK_8X4 as usize] = Some(rav1e_satd_8x4_avx2);
+  out[BLOCK_8X16 as usize] = Some(rav1e_satd_8x16_avx2);
+  out[BLOCK_16X8 as usize] = Some(rav1e_satd_16x8_avx2);
+  out[BLOCK_16X32 as usize] = Some(rav1e_satd_16x32_avx2);
+  out[BLOCK_32X16 as usize] = Some(rav1e_satd_32x16_avx2);
+  out[BLOCK_32X64 as usize] = Some(rav1e_satd_32x64_avx2);
+  out[BLOCK_64X32 as usize] = Some(rav1e_satd_64x32_avx2);
+  out[BLOCK_64X128 as usize] = Some(rav1e_satd_64x128_avx2);
+  out[BLOCK_128X64 as usize] = Some(rav1e_satd_128x64_avx2);
 
-    out[BLOCK_4X8 as usize] = Some(rav1e_satd_4x8_avx2);
-    out[BLOCK_8X4 as usize] = Some(rav1e_satd_8x4_avx2);
-    out[BLOCK_8X16 as usize] = Some(rav1e_satd_8x16_avx2);
-    out[BLOCK_16X8 as usize] = Some(rav1e_satd_16x8_avx2);
-    out[BLOCK_16X32 as usize] = Some(rav1e_satd_16x32_avx2);
-    out[BLOCK_32X16 as usize] = Some(rav1e_satd_32x16_avx2);
-    out[BLOCK_32X64 as usize] = Some(rav1e_satd_32x64_avx2);
-    out[BLOCK_64X32 as usize] = Some(rav1e_satd_64x32_avx2);
-    out[BLOCK_64X128 as usize] = Some(rav1e_satd_64x128_avx2);
-    out[BLOCK_128X64 as usize] = Some(rav1e_satd_128x64_avx2);
+  out[BLOCK_4X16 as usize] = Some(rav1e_satd_4x16_avx2);
+  out[BLOCK_16X4 as usize] = Some(rav1e_satd_16x4_avx2);
+  out[BLOCK_8X32 as usize] = Some(rav1e_satd_8x32_avx2);
+  out[BLOCK_32X8 as usize] = Some(rav1e_satd_32x8_avx2);
+  out[BLOCK_16X64 as usize] = Some(rav1e_satd_16x64_avx2);
+  out[BLOCK_64X16 as usize] = Some(rav1e_satd_64x16_avx2);
 
-    out[BLOCK_4X16 as usize] = Some(rav1e_satd_4x16_avx2);
-    out[BLOCK_16X4 as usize] = Some(rav1e_satd_16x4_avx2);
-    out[BLOCK_8X32 as usize] = Some(rav1e_satd_8x32_avx2);
-    out[BLOCK_32X8 as usize] = Some(rav1e_satd_32x8_avx2);
-    out[BLOCK_16X64 as usize] = Some(rav1e_satd_16x64_avx2);
-    out[BLOCK_64X16 as usize] = Some(rav1e_satd_64x16_avx2);
+  out
+};
 
-    out
-  };
+pub(crate) static SATD_FNS: [[Option<SatdFn>; DIST_FNS_LENGTH];
+  CpuFeatureLevel::len()] = {
+  let mut out = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
 
-  pub(crate) static SATD_FNS: [[Option<SatdFn>; DIST_FNS_LENGTH];
-    CpuFeatureLevel::len()] = {
-    let mut out = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::AVX2 as usize] = SATD_FNS_AVX2;
 
-    out[CpuFeatureLevel::AVX2 as usize] = SATD_FNS_AVX2;
+  out
+};
 
-    out
-  };
-
-  pub(crate) static SATD_HBD_FNS: [[Option<SatdHBDFn>; DIST_FNS_LENGTH];
-    CpuFeatureLevel::len()] =
-    [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];
-}
+pub(crate) static SATD_HBD_FNS: [[Option<SatdHBDFn>; DIST_FNS_LENGTH];
+  CpuFeatureLevel::len()] = [[None; DIST_FNS_LENGTH]; CpuFeatureLevel::len()];

--- a/src/asm/x86/mc.rs
+++ b/src/asm/x86/mc.rs
@@ -9,12 +9,10 @@
 
 use crate::cpu_features::CpuFeatureLevel;
 use crate::frame::*;
+use crate::mc::FilterMode::*;
 use crate::mc::*;
 use crate::tiling::*;
 use crate::util::*;
-
-#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
-use x86_64::*;
 
 type PutFn = unsafe extern fn(
   dst: *mut u8,
@@ -263,111 +261,105 @@ pub fn mc_avg<T: Pixel>(
   }
 }
 
-#[cfg(all(target_arch = "x86_64", feature = "nasm"))]
-mod x86_64 {
-  use self::FilterMode::*;
-  use super::*;
-
-  macro_rules! decl_mc_fns {
-    ($(($mode_x:expr, $mode_y:expr, $func_name:ident)),+) => {
-      extern {
-        $(
-          fn $func_name(
-            dst: *mut u8, dst_stride: isize, src: *const u8, src_stride: isize,
-            w: i32, h: i32, mx: i32, my: i32
-          );
-        )*
-      }
-
-      static PUT_FNS_AVX2: [Option<PutFn>; 16] = {
-        let mut out: [Option<PutFn>; 16] = [None; 16];
-        $(
-          out[get_2d_mode_idx($mode_x, $mode_y)] = Some($func_name);
-        )*
-        out
-      };
+macro_rules! decl_mc_fns {
+  ($(($mode_x:expr, $mode_y:expr, $func_name:ident)),+) => {
+    extern {
+      $(
+        fn $func_name(
+          dst: *mut u8, dst_stride: isize, src: *const u8, src_stride: isize,
+          w: i32, h: i32, mx: i32, my: i32
+        );
+      )*
     }
+
+    static PUT_FNS_AVX2: [Option<PutFn>; 16] = {
+      let mut out: [Option<PutFn>; 16] = [None; 16];
+      $(
+        out[get_2d_mode_idx($mode_x, $mode_y)] = Some($func_name);
+      )*
+      out
+    };
   }
-
-  decl_mc_fns!(
-    (REGULAR, REGULAR, rav1e_put_8tap_regular_avx2),
-    (REGULAR, SMOOTH, rav1e_put_8tap_regular_smooth_avx2),
-    (REGULAR, SHARP, rav1e_put_8tap_regular_sharp_avx2),
-    (SMOOTH, REGULAR, rav1e_put_8tap_smooth_regular_avx2),
-    (SMOOTH, SMOOTH, rav1e_put_8tap_smooth_avx2),
-    (SMOOTH, SHARP, rav1e_put_8tap_smooth_sharp_avx2),
-    (SHARP, REGULAR, rav1e_put_8tap_sharp_regular_avx2),
-    (SHARP, SMOOTH, rav1e_put_8tap_sharp_smooth_avx2),
-    (SHARP, SHARP, rav1e_put_8tap_sharp_avx2),
-    (BILINEAR, BILINEAR, rav1e_put_bilin_avx2)
-  );
-
-  pub(crate) static PUT_FNS: [[Option<PutFn>; 16]; CpuFeatureLevel::len()] = {
-    let mut out = [[None; 16]; CpuFeatureLevel::len()];
-    out[CpuFeatureLevel::AVX2 as usize] = PUT_FNS_AVX2;
-    out
-  };
-
-  pub(crate) static PUT_HBD_FNS: [[Option<PutHBDFn>; 16];
-    CpuFeatureLevel::len()] = [[None; 16]; CpuFeatureLevel::len()];
-
-  macro_rules! decl_mct_fns {
-    ($(($mode_x:expr, $mode_y:expr, $func_name:ident)),+) => {
-      extern {
-        $(
-          fn $func_name(
-            tmp: *mut i16, src: *const u8, src_stride: libc::ptrdiff_t, w: i32,
-            h: i32, mx: i32, my: i32
-          );
-        )*
-      }
-
-      static PREP_FNS_AVX2: [Option<PrepFn>; 16] = {
-        let mut out: [Option<PrepFn>; 16] = [None; 16];
-        $(
-          out[get_2d_mode_idx($mode_x, $mode_y)] = Some($func_name);
-        )*
-        out
-      };
-    }
-  }
-
-  decl_mct_fns!(
-    (REGULAR, REGULAR, rav1e_prep_8tap_regular_avx2),
-    (REGULAR, SMOOTH, rav1e_prep_8tap_regular_smooth_avx2),
-    (REGULAR, SHARP, rav1e_prep_8tap_regular_sharp_avx2),
-    (SMOOTH, REGULAR, rav1e_prep_8tap_smooth_regular_avx2),
-    (SMOOTH, SMOOTH, rav1e_prep_8tap_smooth_avx2),
-    (SMOOTH, SHARP, rav1e_prep_8tap_smooth_sharp_avx2),
-    (SHARP, REGULAR, rav1e_prep_8tap_sharp_regular_avx2),
-    (SHARP, SMOOTH, rav1e_prep_8tap_sharp_smooth_avx2),
-    (SHARP, SHARP, rav1e_prep_8tap_sharp_avx2),
-    (BILINEAR, BILINEAR, rav1e_prep_bilin_avx2)
-  );
-
-  pub(crate) static PREP_FNS: [[Option<PrepFn>; 16]; CpuFeatureLevel::len()] = {
-    let mut out = [[None; 16]; CpuFeatureLevel::len()];
-    out[CpuFeatureLevel::AVX2 as usize] = PREP_FNS_AVX2;
-    out
-  };
-
-  pub(crate) static PREP_HBD_FNS: [[Option<PrepHBDFn>; 16];
-    CpuFeatureLevel::len()] = [[None; 16]; CpuFeatureLevel::len()];
-
-  extern {
-    fn rav1e_avg_avx2(
-      dst: *mut u8, dst_stride: libc::ptrdiff_t, tmp1: *const i16,
-      tmp2: *const i16, w: i32, h: i32,
-    );
-  }
-
-  pub(crate) static AVG_FNS: [Option<AvgFn>; CpuFeatureLevel::len()] = {
-    let mut out: [Option<AvgFn>; CpuFeatureLevel::len()] =
-      [None; CpuFeatureLevel::len()];
-    out[CpuFeatureLevel::AVX2 as usize] = Some(rav1e_avg_avx2);
-    out
-  };
-
-  pub(crate) static AVG_HBD_FNS: [Option<AvgHBDFn>; CpuFeatureLevel::len()] =
-    [None; CpuFeatureLevel::len()];
 }
+
+decl_mc_fns!(
+  (REGULAR, REGULAR, rav1e_put_8tap_regular_avx2),
+  (REGULAR, SMOOTH, rav1e_put_8tap_regular_smooth_avx2),
+  (REGULAR, SHARP, rav1e_put_8tap_regular_sharp_avx2),
+  (SMOOTH, REGULAR, rav1e_put_8tap_smooth_regular_avx2),
+  (SMOOTH, SMOOTH, rav1e_put_8tap_smooth_avx2),
+  (SMOOTH, SHARP, rav1e_put_8tap_smooth_sharp_avx2),
+  (SHARP, REGULAR, rav1e_put_8tap_sharp_regular_avx2),
+  (SHARP, SMOOTH, rav1e_put_8tap_sharp_smooth_avx2),
+  (SHARP, SHARP, rav1e_put_8tap_sharp_avx2),
+  (BILINEAR, BILINEAR, rav1e_put_bilin_avx2)
+);
+
+pub(crate) static PUT_FNS: [[Option<PutFn>; 16]; CpuFeatureLevel::len()] = {
+  let mut out = [[None; 16]; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::AVX2 as usize] = PUT_FNS_AVX2;
+  out
+};
+
+pub(crate) static PUT_HBD_FNS: [[Option<PutHBDFn>; 16];
+  CpuFeatureLevel::len()] = [[None; 16]; CpuFeatureLevel::len()];
+
+macro_rules! decl_mct_fns {
+  ($(($mode_x:expr, $mode_y:expr, $func_name:ident)),+) => {
+    extern {
+      $(
+        fn $func_name(
+          tmp: *mut i16, src: *const u8, src_stride: libc::ptrdiff_t, w: i32,
+          h: i32, mx: i32, my: i32
+        );
+      )*
+    }
+
+    static PREP_FNS_AVX2: [Option<PrepFn>; 16] = {
+      let mut out: [Option<PrepFn>; 16] = [None; 16];
+      $(
+        out[get_2d_mode_idx($mode_x, $mode_y)] = Some($func_name);
+      )*
+      out
+    };
+  }
+}
+
+decl_mct_fns!(
+  (REGULAR, REGULAR, rav1e_prep_8tap_regular_avx2),
+  (REGULAR, SMOOTH, rav1e_prep_8tap_regular_smooth_avx2),
+  (REGULAR, SHARP, rav1e_prep_8tap_regular_sharp_avx2),
+  (SMOOTH, REGULAR, rav1e_prep_8tap_smooth_regular_avx2),
+  (SMOOTH, SMOOTH, rav1e_prep_8tap_smooth_avx2),
+  (SMOOTH, SHARP, rav1e_prep_8tap_smooth_sharp_avx2),
+  (SHARP, REGULAR, rav1e_prep_8tap_sharp_regular_avx2),
+  (SHARP, SMOOTH, rav1e_prep_8tap_sharp_smooth_avx2),
+  (SHARP, SHARP, rav1e_prep_8tap_sharp_avx2),
+  (BILINEAR, BILINEAR, rav1e_prep_bilin_avx2)
+);
+
+pub(crate) static PREP_FNS: [[Option<PrepFn>; 16]; CpuFeatureLevel::len()] = {
+  let mut out = [[None; 16]; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::AVX2 as usize] = PREP_FNS_AVX2;
+  out
+};
+
+pub(crate) static PREP_HBD_FNS: [[Option<PrepHBDFn>; 16];
+  CpuFeatureLevel::len()] = [[None; 16]; CpuFeatureLevel::len()];
+
+extern {
+  fn rav1e_avg_avx2(
+    dst: *mut u8, dst_stride: libc::ptrdiff_t, tmp1: *const i16,
+    tmp2: *const i16, w: i32, h: i32,
+  );
+}
+
+pub(crate) static AVG_FNS: [Option<AvgFn>; CpuFeatureLevel::len()] = {
+  let mut out: [Option<AvgFn>; CpuFeatureLevel::len()] =
+    [None; CpuFeatureLevel::len()];
+  out[CpuFeatureLevel::AVX2 as usize] = Some(rav1e_avg_avx2);
+  out
+};
+
+pub(crate) static AVG_HBD_FNS: [Option<AvgHBDFn>; CpuFeatureLevel::len()] =
+  [None; CpuFeatureLevel::len()];

